src/cookie.ts

```diff
@@ -1,53 +1,76 @@
 export type CookieOpts = {
-  maxAge?: number;
-  domain?: string;
-  path?: string;
-  expires?: Date;
-  httpOnly?: boolean;
-  secure?: boolean;
-  sameSite?: "Lax" | "Strict" | "None" | true;
-  crossSubdomain?: boolean;
+    maxAge?: number;
+    domain?: string;
+    path?: string;
+    expires?: Date;
+    httpOnly?: boolean;
+    secure?: boolean;
+    sameSite?: "Lax" | "Strict" | "None" | true;
+    crossSubdomain?: boolean;
 };
 const DOMAIN_MATCH_REGEX = /[a-z0-9][a-z0-9-]+\.[a-z.]{2,6}$/i
 
+const getCrossSubdomain = () => {
+    const matches = document.location.hostname.match(DOMAIN_MATCH_REGEX)
+
+    const domain = matches ? matches[0] : ''
+
+    return domain ? `.${domain}` : ''
+}
+
 // Note: updated this method to test on staging (Ref:: https://github.com/PostHog/posthog-js/blob/master/src/storage.ts#L42)
+// Commented out some jitsu cookies setters that are not bring used in posthog-js
 export function serializeCookie(name, val, opt: CookieOpts = {}) {
-  const {
-    crossSubdomain,
-    expires: date,
-    secure: isSecure,
-    domain
-  } = opt;
-
-  try {
-    let cdomain = '',
-      expires = '',
-      secure = ''
-
-    if (crossSubdomain) {
-      const matches = document.location.hostname.match(DOMAIN_MATCH_REGEX),
-        domain = matches ? matches[0] : ''
-
-      cdomain = domain ? '; domain=.' + domain : ''
-    } else {
-      if (domain) {
-        cdomain = '; domain=' + domain
-      }
-    }
+    try {
+        const {
+            maxAge,
+            domain,
+            path,
+            expires,
+            httpOnly,
+            secure,
+            sameSite,
+            crossSubdomain,
+        } = opt
 
-    if (date) {
-      expires = '; expires=' + date.toUTCString()
-    }
+        let new_cookie_val = `${name}=${encodeURIComponent(val)}`
 
-    if (isSecure) {
-      secure = '; secure'
-    }
+        if (crossSubdomain) {
+            const cross_subdomain = getCrossSubdomain()
+            new_cookie_val += `; domain=${cross_subdomain}`
+        } else if (domain) {
+            new_cookie_val += `; domain=${domain}`
+        }
+
+        if (path) {
+            new_cookie_val += `; path=${path}`
+        } else {
+            new_cookie_val += `; path=/`
+        }
 
-    const new_cookie_val =
-      name + '=' + encodeURIComponent(JSON.stringify(val)) + expires + '; path=/' + cdomain + secure
-    document.cookie = new_cookie_val
-    return new_cookie_val
-  } catch (err) {
-    return
-  }
+        if (expires) {
+            new_cookie_val += `; expires=${expires.toUTCString()}`
+        }
+
+        // if (maxAge) {
+        //     new_cookie_val += `; max-age=${maxAge}`
+        // }
+
+        // if (httpOnly) {
+        //     new_cookie_val += `; httponly`
+        // }
```

Comment:
Commented out `httpOnly`, `maxAge` and `sameSite` for testing as they are not implemented in posthog cookies setting implementation